ee/vellum_ee/workflows/display/tests/workflow_serialization/test_scheduled_trigger_serialization.py

```diff
@@ -0,0 +1,145 @@
+    with mock_metadata_json(
+        "test_scheduled_trigger_serialization_with_metadata_json.<locals>.DailyScheduleTrigger",
+        expected_trigger_id,
+    ):
+        # Define the trigger class AFTER setting up mocks
+        # This ensures the metaclass reads from our mocked metadata.json
+        class DailyScheduleTrigger(ScheduleTrigger):
+            class Config:
+                cron = "0 9 * * *"  # Every day at 9am
+                timezone = "America/New_York"
+
+        class ProcessNode(BaseNode):
+            class Outputs(BaseNode.Outputs):
+                timestamp = DailyScheduleTrigger.current_run_at
+
+            def run(self) -> Outputs:
+                return self.Outputs()
+
+        # AND a workflow that has the scheduled trigger as the entrypoint
+        class TestWorkflow(BaseWorkflow[BaseInputs, BaseState]):
+            graph = DailyScheduleTrigger >> ProcessNode
+
+        # WHEN we serialize the workflow
+        # The trigger ID should be automatically loaded from metadata.json by the metaclass
+        result: dict = get_workflow_display(workflow_class=TestWorkflow).serialize()
+
+        # THEN we get the expected trigger
+        assert "triggers" in result
+        triggers = result["triggers"]
+        assert isinstance(triggers, list)
+        assert len(triggers) == 1
+
+        trigger = triggers[0]
+        assert isinstance(trigger, dict)
+        assert trigger["type"] == "SCHEDULED"
+
+        # AND the id is the one from metadata.json
+        assert trigger["id"] == str(expected_trigger_id)
+
+
+def test_scheduled_trigger_serialization_without_metadata_json():
+    """ScheduleTrigger uses default hash-based __id__ when no metadata.json is available."""
+
+    # GIVEN a scheduled trigger defined without metadata.json
+    class WeeklyScheduleTrigger(ScheduleTrigger):
+        class Config:
+            cron = "0 9 * * MON"  # Every Monday at 9am
+            timezone = "UTC"
+
+    class ProcessNode(BaseNode):
+        class Outputs(BaseNode.Outputs):
+            timestamp = WeeklyScheduleTrigger.current_run_at
+
+        def run(self) -> Outputs:
+            return self.Outputs()
+
+    # AND a workflow that has the scheduled trigger as the entrypoint
+    class TestWorkflow(BaseWorkflow[BaseInputs, BaseState]):
+        graph = WeeklyScheduleTrigger >> ProcessNode
+
+    # WHEN we serialize the workflow without metadata.json
+    result: dict = get_workflow_display(workflow_class=TestWorkflow).serialize()
+
+    # THEN we get the expected trigger
+    assert "triggers" in result
+    triggers = result["triggers"]
+    assert isinstance(triggers, list)
+    assert len(triggers) == 1
+
+    trigger = triggers[0]
+    assert isinstance(trigger, dict)
+    assert trigger["type"] == "SCHEDULED"
+
+    # AND the id is the trigger class's default __id__ (hash-based)
+    expected_default_id = str(WeeklyScheduleTrigger.__id__)
+    assert trigger["id"] == expected_default_id
```

Comment:
[nit] this assertion would be true even in the metadata.json case bc we are updating the `__id__` here. I would set the expected to be the hardcoded value here